File: Project2/store/server/src/models/book.js
const { PUSHER_CHANNEL_STORE } = require('./../config/configs');
const { messageType } = require('../enums');
const { sendNotificationMessage } = require('../services/websockets/pusher');

module.exports = (sequelize, DataTypes) => {
  const Book = sequelize.define('Book', {
    title: {
      type: DataTypes.STRING,
      unique: true,
      allowNull: false,
    },
    author: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    price: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: false,
      defaultValue: 0,
    },
    stock: {
      type: DataTypes.INTEGER,
      allowNull: false,
      defaultValue: 0,
    },
    // TODO: add images later
  });

  Book.afterUpdate(async (book) => {
    sendNotificationMessage(PUSHER_CHANNEL_STORE, messageType.updateBook, book);
  });

  return Book;
};

File: Project2/store/interface/GUI/Book.cs
namespace @interface
{
    public class Book
    {
        public int id { get; set; }
        public string title { get; set; }
        public string author { get; set; }
        public double price { get; set; }
        public int stock { get; set; }

        public Book()
        {
        }

        public Book(dynamic book)
        {
            id = (int) book.id;
            title = (string) book.title;
            author = (string) book.author;
            price = (double) book.price;
            stock = (int) book.stock;
        }

        public Book(int id, string title, string author, double price, int stock)
        {
            this.id = id;
            this.title = title;
            this.author = author;
            this.price = price;
            this.stock = stock;
        }
    }
}